src/Components/Line.js

```diff
@@ -1,47 +1,99 @@
 import { useEffect, useState } from "react";
-import "./Line.css";
-import Dot from "./Dot.js";
+import styles from "./Line.module.css";
+import DotLine from "./DotLine.js";
+import InputLine from "./InputLine";
+
+const DIGITS = /[0-9]+/;
 const SIZE = 4;
+const intialInputs = Array(SIZE).fill("");
 const Line = (props) => {
-  const checkCodeValidity = () => {};
-  console.log(props); //secret code
+  const [enteredCode, setEnteredCode] = useState(intialInputs);
+  const [circles, setCircles] = useState([]);
+  //to restart the game and clear inputs
+  useEffect(() => {
+    setEnteredCode(intialInputs);
+    setCircles([]);
+  }, [props.clear]);
 
-  // const [enteredValue, setEnteredValue] = useState([0, 0, 0, 0]);
-  const [enteredValue, setEnteredValue] = useState(["", "", "", ""]);
+  useEffect(() => {});
+  //check if input is number & add into the array
   const valueInputChangeHandler = (digitValue, index) => {
-    const clonedValues = [...enteredValue];
-    clonedValues[index] = digitValue;
+    if (DIGITS.test(digitValue) === false) {
+      console.log(DIGITS.test(digitValue));
+      return;
+    }
+    const tempEnteredCode = [...enteredCode];
+
+    tempEnteredCode[index] = digitValue;
+    setEnteredCode(tempEnteredCode);
+  };
+
+  //check button
+  const checkCodeValidity = () => {
+    let stepCount = props.activeStep;
+    let arr = [];
+
+    for (let i in props.secretCode) {
+      const enteredDigit = parseInt(enteredCode[i]);
+
+      //search for entered digit in the secretcode
+      let searchInSecretCode = props.secretCode.findIndex(
+        (val) => val === enteredDigit
+      );
 
-    setEnteredValue(clonedValues);
-    //  setEnteredValue(prev)
+      //search for the digit in the secretcode
+      let searching = props.secretCode.findIndex(
+        (val) => val === props.secretCode[i]
+      );
+
+      if (searchInSecretCode !== -1 && searching === searchInSecretCode) {
+        arr.push(true);
+      } else if (
+        searchInSecretCode !== -1 &&
+        searching !== searchInSecretCode
+      ) {
+        arr.push(false);
+      }
+    }
+    //sort boolean array
+    arr.sort((value) => {
+      return value ? -1 : 1;
+    });
+    setCircles(arr);
+
+    if (arr.every((val) => val !== false) && arr.length === 4) {
+      props.updateIsAWin(true);
+    } else if (props.activeStep === 7) {
```

Comment:
use the const that you defined in the Game component (gameSize)

Reply:
done